common/oscanner.cpp

```diff
@@ -326,4 +332,146 @@ bool OScanner::isQuotedString() const
 	return _isQuotedString;
 }
 
+//
+// Get token as an int.
+//
+int OScanner::getTokenAsInt() const
+{
+	// fix for parser reading in commas
+	std::string str = _token;
+
+	// remove comma if necessary
+	if (str[str.length() - 1] == ',')
+	{
+		str[str.length() - 1] = '\0';
+	}
+
+	char* stopper;
+
+	if (str == "MAXINT")
+	{
+		return MAXINT;//INT32_MAX;
+	}
+
+	const int num = strtol(str.c_str(), &stopper, 0);
+
+	if (*stopper != 0)
+	{
+		I_Error("Bad numeric constant \"%s\".", str.c_str());
+	}
+
+	return num;
+}
+
+//
+// Get token as a float.
+//
+float OScanner::getTokenAsFloat() const
+{
+	// fix for parser reading in commas
+	std::string str = _token;
+
+	// remove comma if necessary
+	if (str[str.length() - 1] == ',')
+	{
+		str[str.length() - 1] = '\0';
+	}
+
+	char* stopper;
+
+	const double num = strtod(str.c_str(), &stopper);
+
+	if (*stopper != 0)
+	{
+		I_Error("Bad numeric constant \"%s\".", str.c_str());
+	}
+
+	return static_cast<float>(num);
+}
+
+//
+// Get token as a bool.
+//
+bool OScanner::getTokenAsBool() const
+{
+	return stricmp(_token.c_str(), "true") == 0;
```

Comment:
We have a function specifically for dealing with std::string case-insensitive comparisons:

https://github.com/odamex/odamex/blob/stable/common/cmdlib.cpp#L240